packages/components/src/autocomplete/index.tsx

```diff
@@ -385,7 +338,7 @@ function useAutocomplete( {
 	const isExpanded = !! autocompleter && filteredOptions.length > 0;
 	const listBoxId = isExpanded
 		? `components-autocomplete-listbox-${ instanceId }`
-		: null;
+		: undefined;
```

Comment:
Runtime change. This prop is passed to `AutocompleterUI` (below), which renders a `Popover` component. That `Popover` component expects `listBoxId` to be `string | undefined`, but not `null` because it's used as a `div`'s `id`.